YatimaStdLib/ByteArray.lean

```diff
@@ -35,4 +35,73 @@ instance : Ord ByteArray where
 def Subarray.asBA (s : Subarray UInt8) : ByteArray :=
   s.as.data.toByteArray
 
+def UInt8.showBits (u : UInt8) : String :=
+  let numStr := u.toNat |> Nat.toDigits 2
+  "".pushn '0' (8 - numStr.length) ++ ⟨numStr⟩
+
+def ByteArray.toString (bs : ByteArray) : String := Id.run do
+  if bs.isEmpty then "b[]" else
+  let mut ans := "b["
+  for u in bs do
+    ans := ans ++ UInt8.showBits u ++ ","
+  return ans.dropRight 1 ++ "]"
+
+instance : Repr ByteArray where
+  reprPrec bs _ := toString bs
+
+def ByteArray.padLeft (bs : ByteArray) (u : UInt8) : Nat → ByteArray
+  | 0 => bs
+  | n + 1 => ByteArray.mk #[u] ++ padLeft bs u n
+
+partial def UInt64.toByteArray (u : UInt64) : ByteArray :=
+  let rec loop (u : UInt64) (acc : Array UInt8) :=
+    if u == 0 then acc else loop (u >>> 8) <| acc.push (u &&& (0xff : UInt64)).toUInt8
+  loop u #[] |>.reverse
+             |> ByteArray.mk
+             |> (fun bs => ByteArray.padLeft bs 0 (8 - bs.size))
+inductive Bit where
+  | one
+  | zero
+deriving BEq
+
+instance : ToString Bit where
+  toString
+    | .one  => "1"
+    | .zero => "0"
+open Bit
+
+def Bit.xOr : Bit → Bit → Bit
+  | one, zero
+  | zero, one => one
+  | _, _      => zero
+
+def Bit.toNat : Bit → Nat
+  | zero => 0
+  | one  => 1
+
+def Bit.toUInt8 : Bit → UInt8 := Nat.toUInt8 ∘ Bit.toNat
+def bArXOr (bs : Array Bit) : Bit := bs.foldl (fun b b' => b.xOr b') zero
+
+def bArToNat (bs : Array Bit) : Nat := bs.foldl (fun b b' => b*2 + b'.toNat) 0
+
+def ByteArray.getD (bs : ByteArray) (idx : Int) : UInt8 :=
```

Comment:
This function is a bit weird. Why not defer to `Array.getD`? The standard for `getD` should receive a default value for when `get?` returns `none`. And it's still using the panicking notation